src/saltext/vmware/modules/license_mgr.py

```diff
@@ -0,0 +1,207 @@
+            ret["licenses"] = license_key
+            ret["message"] = "Test dry-run, not really connected to a vCenter testing"
+            return ret
+
+        result = utils_license_mgr.add_license(
+            service_instance, license_key, datacenter_name, cluster_name, esxi_hostname
+        )
+        if result:
+            ret["licenses"] = license_key
+
+    except (
+        salt.exceptions.VMwareApiError,
+        salt.exceptions.VMwareObjectRetrievalError,
+        salt.exceptions.VMwareRuntimeError,
+        salt.exceptions.CommandExecutionError,
+    ) as exc:
+        log.exception(exc)
+        ret[
+            "message"
+        ] = f"Failed to add a license key '{license_key}' due to Exception '{str(exc)}'"
+        ret["result"] = False
+        return ret
+
+    if not result:
+        ret[
+            "message"
+        ] = f"Failed specified license key '{license_key}' was not added to License Manager"
+        ret["result"] = False
+
+    return ret
+
+
+def list_(service_instance=None):
+    """
+    Returns a list of licenses for the specified Service Instance
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one [default None]
+
+    .. code-block:: bash
+
+        salt '*' vmware_license_mgr.list
+    """
+    log.debug("start vmware ext license_mgr list")
+    if service_instance is None:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+
+    if "test" in __opts__ and __opts__["test"]:
+        ret["licenses"] = "DGMTT-FAKED-TESTS-LICEN-SE012"
+        ret["message"] = "Test dry-run, not really connected to a vCenter testing"
+        return ret
+
+    return utils_license_mgr.list_licenses(service_instance)
+
+
+def remove(license_key, **kwargs):
+    """
+    Remove a license specified by license_key from a Datacenter, Cluster, ESXI Server or vCenter
+    If no datacenter, cluster or ESXI Server is specified, it is assumed the operation is to be applied to a vCenter
+
+    license_key
+        License Key which specifies license to remove from the license manager
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one [default None]
+
+    datacenter_name
+        Datacenter name to use for the operation [default None]
+
+    cluster_name
+        Name of the cluster to add license [default None]
+
+    esxi_hostname
+        Hostname of the ESXI Server to add license [default None]
+
+    CLI Example:
+
+    .. code-block: bash
+
+        salt '*' vmware_license_mgr.remove license_key
+    """
+    log.debug("start vmware ext license_mgr remove license_key")
+    ret = {}
+    op = {}
+    for key, value in kwargs.items():
+        op[key] = value
+
+    service_instance = op.pop("service_instance", None)
+    datacenter_name = op.pop("datacenter_name", None)
+    cluster_name = op.pop("cluster_name", None)
+    esxi_hostname = op.pop("esxi_hostname", None)
```

Comment:
C & P from previously handling kwargs in salt/modules/junos.py where had to be real picky about what was accepted as keywords.  Some valid keywords when used in a slightly different command would cause the Juniper router to barf.

Will change to what you suggest